scalalib/src/ScalaModule.scala

```diff
@@ -169,34 +172,34 @@ trait ScalaModule extends JavaModule { outer =>
     os.makeDir.all(javadocDir)
 
     if (isDotty(scalaVersion())) {
-      val root = dottydocSiteRoot().path
-      if (os.exists(root)) {
-        os.copy(root, javadocDir, replaceExisting = true)
-      } else {
-        os.makeDir.all(javadocDir)
+      if (docSources().length > 1) {
+        T.ctx.log.error(
```

Comment:
We could actually handle this situation by copying the content of all dirs into a new temporary location and using this instead.

Reply:
I thought of that, but then I also thought that it may not be worth the added complexity. Ideally dottydoc would actually support specifying multiple inputs (and also make the output path configurable), then we wouldn't need to handle merging directories.

In case you do think it's worthwhile to merge many input dirs, then I'd suggest we try a symlink based approach, where for every source directoy we symlink all contents into the final source dir (latter overwriting the former).

Reply:
Wdyt of something like
```
if (isDotty(scalaVersion())) {
      for {
        ref <- docSources()
        path = ref.path
        if os.exists(path) && os.isDir(path)
        children = os.list(path)
        src <- children
      } {
        val dest = javadocDir / src.last
        if (os.exists(dest)) os.remove(dest)
        os.symlink(dest, src)
      }
    }
```
?

Reply:
added that in the latest commit

Reply:
I'm not sure if symlinks will work in all cases, e.g. Windows with FAT filesystem, if that is still a thing. Also, we should probably avoid that step, if we only have one exsisting docSources.

Reply:
Let's remove that complexity then. Ideally, we could use just `T.source`, but that removes the optional nature. We can print the error in case we found more than one _existing_ docSource dir and the user can still work around by creating some merge targets themselves.

Reply:
:disappointed: oh well, I'll do a walk and copy then

Reply:
Up to you, I can also just walk all children and copy them instead of symlinking.

Reply:
Ok. If more than one dir _exists_, copy them into a merge directory. Otherwise just use the original docSources. A `T.log.debug`  message when using the synthetic merge dir would be nice.

Reply:
Unfortunately we need the copy even for single docSources, as dottydoc (as it currently is) will will pollute the input directory with its output directory _site.

Reply:
Well, strictly speaking, we don't *need* the copy. However, if we don't then we'll modify something outside of the tasks `dest` dir. Not sure if that's acceptable?

Reply:
Fine. But then merging is the most natural behaviour anyway.

Reply:
Ok, sounds good! I'll go ahead and implement the merging

Reply:
I also added a test to check that merging of multiple doc directories works as expected